students/stefanjp1/lesson03/mailroom.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+def donor_names():
+    """Return a list of donor names."""
+    names = list()
+    for name in donor_db:
+        names = names + [name[0]]
+    return names
+
+
+def send_thank_you():
+    """Create thank you card formatting."""
+    full_name = input("Enter the donor's full name > ")
+    
+    donors = donor_names()
+    
+    if full_name == 'list':
+        #Print the donor names and restart the function
+        for name in donors:
+            print(name)
+            
+        send_thank_you()
+    
+    if full_name not in donors:
+        donor_db.append([(full_name)])
+    
+    amount = float( input("Enter the donation amount > ") )
+    
+    for index, row in enumerate(donor_db):
+        if row[0] == full_name:
+            new_row = row + [amount]
+            
+            donor_db.pop(index)
+            donor_db.append((new_row,))
```

Comment:
if you found donor in existing db, then you should just append to it to simplify:
```
for row in donor_db:
    if row[0] == full_name:
        row.append(amount)
        break
else:
    # add new donor here 
```
you can also add new donor in the else clause.